### ISSUE TITLE
Even simpler data upload.

### ISSUE THREAD

**Issue by alexey-milovidov** [ClickHouse/ClickHouse#38775]
**Use case**

In the case of HTTP PUT request and POST with form file upload, if the query is not specified, we can assume that it is INSERT into a table with the name as the corresponding file name, with format autodetected from the file name.

If the filename contains extra components, the first component can be interpreted as a database name.

There should be a setting to determine what to do if a table does not exist: - throw an exception (by default); create a table with the default engine, using schema inference from the data; create or replace a table; - or the opposite - throw an exception if table exists.

**Additional context**

We already allowing to configure custom HTTP API endpoints for INSERTs.

We can also allow to put the database name and table name in the path like this:
```
http://localhost:8123/db/table
```

And it makes sense also for the existing API.

**Comment by royxact** [comment]
Hi! any updates on this? would be very useful

**Comment by alexey-milovidov** [comment]
@royxact This task is not in progress. It is considered for implementation, but we did not start implementing it.

**Comment by pkit** [comment]
@alexey-milovidov apropos "simple inserts" 
Right now the current **predefined** HTTP interface cannot read the HTTP request body as a payload.
I've had a simple patch that supports it, should I post it as a PR?

Example:
```xml
<http_handlers>
    <rule>
        <url>/table1</url>
        <methods>POST</methods>
        <handler>
            <type>predefined_query_handler</type>
            <query>INSERT INTO table1 FORMAT JsonEachRow</query>
            <useRequestBody/>
        </handler>
    </rule>
    <defaults/>
 ```

**Comment by alexey-milovidov** [comment]
@pkit Yes, I expected it should already work this way, but I never tried. Thank you!

**Comment by pkit** [comment]
@alexey-milovidov it doesn't really work, because of https://github.com/ClickHouse/ClickHouse/issues/7488 as the body is demangled into oblivion.
I had some hacks to support it, but now I need to make them more generic, to be PR-able as a feature.
Will update when I'm ready.

**Comment by jdbohrman** [comment]
@alexey-milovidov Would you like me and @OscarBrad to look into this?

**Comment by Lrabbit125** [comment]
@alexey-milovidov  Hi there, could you please assign this task to me? I've been working on it recently, and I have experience with similar issues in other OLAP engines like Doris and StarRocks. I'll keep you updated on any progress.

**Comment by alexey-milovidov** [comment]
@Lrabbit125, thanks, it will be very appreciated!

**Comment by Felixoid** [comment]
I am working on a similar task currently, and concluded that a new SQL function `HMAC(message, key, algorythm)` coould help here.

For example, the data column's contstraint compared as `concat('sha256=', HMAC(data, 'some-very-sercret-key', 'sha256')) == getClientHTTPHeader('X-Hub-Signature-256')`

Would it work? If yes, I can try to implement this function.

**Comment by alexey-milovidov** [comment]
@Felixoid, makes sense, and it will be useful! Let's make the arguments order similar to `encrypt`.

**Comment by alexey-milovidov** [comment]
However, why do we need it in this context? For web hooks, you just authenticate them with any of the methods (HTTP basic, URL parameters, HTTP headers, ...), and they do regular INSERTs.

**Comment by Felixoid** [comment]
Kind of opposite. Webhooks don't use any authentication, but validate integrity by the HMAC. See [github webhooks](https://docs.github.com/en/webhooks/using-webhooks/validating-webhook-deliveries#validating-webhook-deliveries) for example

**Comment by alexey-milovidov** [comment]
@Felixoid, you can embed auth into the URL and skip the HMAC validation for this case.

**Comment by Felixoid** [comment]
I know, but as you can see in https://github.com/ClickHouse/ClickHouse/issues/73900 as well, HMAC is the pretty common thing for webhooks.

It's not a blocker for me, rather a function to comply the recommended workflow.

**Comment by AJ-here** [comment]
Hi @alexey-milovidov 

I'm currently learning ClickHouse and exploring its internals, and I'd like to start contributing to the project. I came across this warmup task and would be interested in working on it.

Since the issue covers several behaviors, I was thinking of starting with a smaller first step: supporting HTTP PUT requests to `/database/table` without an explicit query by constructing the corresponding INSERT query and reusing the existing input-format detection infrastructure where possible.

Before starting the implementation, I'd like to trace the current HTTP request handling path and confirm whether this incremental scope still aligns with the intended direction of the issue.

Is this issue still available for contribution, and would a PR implementing this smaller first step be acceptable?

Thanks!

**Comment by Felixoid** [comment]
I think, #110312 is related to this as well.
